### DISCUSSION TITLE
Daily flow

### DISCUSSION THREAD

**Discussion by jrock2004** [nicknisi/dotfiles#171] [Q&A]
Taking a look at your screenshot, I am curious if this is how you work, day to day? Do you just have 1  tmux session and use tmux tab/screens for different items?

![](https://github.com/nicknisi/dotfiles/raw/master/screenshot.png)

**Comment by nicknisi** [comment]
Hey @jrock2004 I thought this would be a fun issue to play around with the new-ish Discussions feature on GitHub.

You are correct. I typically create a single tmux session for the line of work I'm doing. This is typically delineated as "work" and "open-source", so I can completely switch contexts and for the most part leave my tmux setup alone for work. Then, I'll typically try and stick to one tmux window per Git repository that I'm working with. In that window, I'll have a top-half split that pretty much continuously runs vim, with as many splits as needed running across the bottom for running watchers, tests, doing git stuff, etc. I also work on an ultra wide monitor, so occasionally I'll create a right-split on the top to have a wider view of things like test output, for example. Whenever I want to concentrate on just writing code or whatever the output is in any particular split, I'll promote it to full-screen using `<prefix>-z`, and then return it to the split by pressing that again. Here's an annotated screenshot to highlight what I'm trying to say here.

![tmux-workflow](https://user-images.githubusercontent.com/293805/109544539-4e71f280-7a8d-11eb-85a7-245bc2ffeff8.png)

I would be curious to hear about your preferred setup as well, and I'm also happy to answer any other questions you might have!

**Reply by jrock2004** [reply]
I totally forgot about discussions, sorry. I typically create a tmux session for each repo that I could be working on. So I have sessions like, `Dotfiles`, `Work Project 1`, `Fun Projects`.  In each of those session I usually have 3 panes, First pane is for my code, 2nd pane is for running yarn start, and 3rd is for any git stuff I do. I also have a wide screen. I am now thinking about doing what you have here though with the following tmux sessions.

Work
- app 1
- app 2
- dotfiles

Personal
- blog
- anime
- dotfiles

**Comment by jrock2004** [comment]
This is old discussion but do you have a browser and terminal on 1 screen? Do you have 2 browsers and 1 terminal? Maybe 1 browser and 2 terminals?


![window-order-current](https://user-images.githubusercontent.com/655716/210159248-e0c5795a-6d11-4315-950f-0adff21150e6.png)

**Reply by nicknisi** [reply]
So, it really varies for me. But, I typically have a full-screen or near-full-screen terminal on my “middle” monitor and then a daily driver browser on the left monitor and a test browser on the right monitor. I’ll typically hide Slack, email, etc. but when it’s open it’ll be on the left screen. My current setup is burdened with too much space! 😅 
![E2070641-FC59-4DDA-B121-7184BD17ACC6](https://user-images.githubusercontent.com/293805/210188334-59f0a1ea-dc56-4999-9915-d9c6d7d415fe.jpeg)

**Reply by jrock2004** [reply]
Ah I only have 1 monitor but this helps
